portfolio/src/main/client/src/components/sections/comments.ts

```diff
@@ -138,23 +143,78 @@ async function submitForm(
   await loadCurrentPage(commentsEl, pages, limit);
 }
 
+/**
+ * @param text The text to truncate.
+ * @returns `text` truncated to 5 lines or 500 chars, whichever is shorter.
+ */
+function truncateText(text: string) {
+  const truncatedByChars = text.slice(0, 500);
+  const truncatedByNewlines = text.split('\n').slice(0, 5).join('\n');
+
+  if (truncatedByChars.length < truncatedByNewlines.length) {
+    return truncatedByChars;
+  } else {
+    return truncatedByNewlines;
+  }
+}
+
 const Comment = (comment: CommentInfo): HTMLElement => {
   // user supplied strings cannot be interpolated directly to avoid XSS
-  const content: HTMLElement = htmlElement`<div class="content"></div>`;
-  content.innerText = comment.content;
+  const contentEl: HTMLElement = htmlElement`<div class="content"></div>`;
+  contentEl.innerText = comment.content;
+
+  let truncatedContentEl: HTMLElement | null = null;
+  let truncatedContentExpander: HTMLAnchorElement | null = null;
+  const truncatedContent = truncateText(comment.content);
+
+  if (truncatedContent.length < comment.content.length) {
+    truncatedContentEl = document.createElement('div');
+    truncatedContentEl.classList.add('content', 'truncated');
+    truncatedContentEl.innerText = truncatedContent;
+
+    truncatedContentExpander = document.createElement('a');
+    truncatedContentExpander.href = 'javascript:void 0';
+    truncatedContentExpander.classList.add('expander');
+    truncatedContentExpander.innerText = 'see more';
+
+    let isTruncated = true;
+    truncatedContentExpander.addEventListener('click', () => {
+      if (isTruncated) {
+        truncatedContentEl!.replaceWith(contentEl);
+        truncatedContentExpander!.innerText = 'see less';
+      } else {
+        contentEl.replaceWith(truncatedContentEl!);
+        truncatedContentExpander!.innerText = 'see more';
+      }
+
+      isTruncated = !isTruncated;
```

Comment:
nit: it's more readable to have state variables change first, then have their side effects resolve. This should just effectively reverse your if statement above.